x-pack/test/apm_api_integration/tests/services/annotations.spec.ts

```diff
@@ -29,40 +37,62 @@ export default function annotationApiTests({ getService }: FtrProviderContext) {
     );
   }
 
-  function request({ method, url, data }: { method: string; url: string; data?: JsonObject }) {
-    switch (method.toLowerCase()) {
-      case 'get':
-        return supertestRead.get(url).set('kbn-xsrf', 'foo');
-
-      case 'post':
-        return supertestWrite.post(url).send(data).set('kbn-xsrf', 'foo');
+  function createAnnotation(body?: {
+    '@timestamp'?: string;
+    message?: string;
+    tags?: string[];
+    service?: { environment?: string; version: string };
+  }) {
+    return apmApiClient.annotationWriterUser({
+      endpoint: 'POST /api/apm/services/{serviceName}/annotation',
+      params: {
+        path: {
+          serviceName: 'opbeans-java',
+        },
+        body: body as {
+          '@timestamp': string;
+          message?: string;
+          tags?: string[];
+          service: { environment?: string; version: string };
+        },
+      },
+    });
+  }
 
-      default:
-        throw new Error(`Unsupported method ${method}`);
-    }
+  function getAnnotation(query: { start: string; end: string; environment: string }) {
```

Comment:
I think you can do something like `query: RecursivePartial<
      APIClientRequestParamsOf<'GET /api/apm/services/{serviceName}/annotation/search'>['params']['query']
    >`